### DISCUSSION TITLE
Allow Service Invoke src shorthand (similar to events and transitions)

### DISCUSSION THREAD

**Discussion by ghost** [statelyai/xstate#1233] [General]
From the docs: 

> An invocation is defined in a state node's configuration with the invoke property, whose value is an object

`invoke: {
    src: service
  }`

For a service that doesn't need an id and other properties, it should be allowed to be defined as just the value of the src property, like `invoke: service`. This is consistent with the way events and transitions definitions can be done.

**Comment by davidkpiano** [comment]
This will probably be a V5 feature if we decide to do it.

**Comment by CodingDive** [comment]
Just out of curiosity: What kind of services don't have an `onDone` or `onError` handler? Even when invoking a callback as a service, I tend to define `onError`.

**Reply by Andarist** [reply]
It's up to you to implement those handlers and you probably should provide them 90+% of times. 

By default callback services can only crash during initialization and machine services can't ever crash on their own so `onError` could be seen as not needed there, but you can always send error events on your own to trigger those handlers. You can also for example use `escalate` helper to forward a "caught" error from a promise service to a grandparent machine, if your structure looks like this:
- grandparent machine
- that invoked a parent machine
- that invoked a promise service

**Reply by davidkpiano** [reply]
In V5, activity services don't have an `onDone` (but they might have `onError`).
